core/DataAccess/LogQueryBuilder/JoinGenerator.php

```diff
@@ -120,6 +120,11 @@ public function generate()
                     $this->joinString .= ' LEFT JOIN';
                 }
 
+                if (!isset($table['joinOn']) && $this->tables->getLogTable($table['table']) && !empty($availableLogTables)) {
```

Comment:
You can now specify a `from` table eg like this: `array('table' => 'log_visit', 'tableAlias' => 'lv')` or `array('table' => 'log_visit', 'join' => 'RIGHT JOIN')`. The advantage is that the core will still find the correct join while being able to change the join type etc.